notebooks/07_fractional_differencing.ipynb

```diff
@@ -0,0 +1,773 @@
/Hcc8/h6NGj0hzNv/32GyZMmKD2OqVSiZCQEAQGBiIoKAhlZWUYPXo0/Pz8MGXKFJiYmMgRn1qwgwcPwsfHB6dOncKIESPkjkN3cfv2bbz77rvYtm0btLS07llw5yK21Jjy8vLw+eef49NPP0VxcTHmzp2LN954A507d5Y7GhERERERtUwstBNRy5aYmAgvLy+kpaVJBTYdHR3MnDlTmi4iMjISgYGB2LNnD7Kzs+Hu7g4/Pz/MnDkTNjY2csanFqy6uhq9evVC//79uZhmK3D16lW8+uqr+OOPP6Cpqam2MKq2tjYmT56Mn3/+WcaE1BaVl5cjICAAGzduREZGBvz8/LBq1SoumE1ERERERHdioZ2IWq7Dhw9j5syZqKioQE1NjdpzBgYGWLhwIfbt24fr16+jV69eeO655/Dcc8/ByclJpsTUmmzevBmrV6/G1atXOUq1FTl+/DgWLVqE5OTkesV2LopKTaW6uho//vgj3n//fSQlJeHZZ5/F2rVr0bNnT7mjERERERFRy8DFUImo5RFCYMOGDfD29kZZWVm9IjsAVFRU4MSJE5g5cyb++usvxMTEYNWqVSyy0wPJy8vD+vXrsXTpUhbZW5lx48YhNjYWmzZtgomJCXR0dKTnAgICZExGbZmOjg7mzJmDuLg47NmzB7GxsejVqxe8vb1x6dIlueMREREREVELwBHtRI2gvLwcFRUVqK2tRWFhIQCguLhYKhBXV1ejpKTknu8vKSm570J/BgYG0NfXv+fzFhYW0mN9fX0YGBhAU1MTZmZmAABjY2O1YlRLVlxcjNmzZ+PQoUNqo1XvpK2tjQkTJiA4OLgZ01FbsXjxYvz8889ISkqCqamp3HHoIeXm5mLt2rX4/PPPoVQq4eDggOvXr6OwsBBCCJSWlqKqqgpVVVUoLS2V3ldQUIB7/fWnrKwMlZWV9zynkZHRPRdT1tHRgbGxsfSzmZkZNDU1YWhoCD09vXrPU+tVW1uLffv2Yf369YiJicHkyZPx9ttvo3///nJHIyIiIiIieXDqGGpfSktLkZeXh/z8fOTn56OoqAhlZWUoLCxESUkJysrKUFJSgsLCQpSWlt71uZqaGhQXFwP47yKd9ysGtzS6urowMjIC8L/ivLm5OQwNDWFoaAgzMzMYGxvDyMgIRkZGd33O1NQUFhYWsLCwQIcOHaCnp9do+ZKSkvDMM8/g2rVrdx3FfidtbW0oFApYWlo2WgZq+5KTk+Hq6opt27ZhwYIFcsdp18rLy5Gfn4+8vDzk5eWhqKgIpaWlUrtbWlqK0tJS5OfnS49LSkpQUFCA0tJSVFZWoqysDGVlZSgtLYVSqZT7kh6Yubk5NDQ0pMK9qt01NjaGubm59LOJiQnMzMxgZGQktcUmJiYwNTVFhw4d0KFDB1hYWLSaztS2RgiBAwcOYN26dYiKioKvry/effdduLq6yh2NiIiIiIiaFwvt1Hrdvn0b2dnZyMnJQVZWFrKzs9WK6Hd7XFVVVe84qpHfxsbGMDQ0hLGxMczMzKQCs6rgYWhoCBMTE7WR4iYmJtDW1oaenh4MDQ0B/K+AfeeoR1VR5cLcBCEAACAASURBVG7qvv9uioqK7llAunO0vGo0Zt0OAdX7KyoqUF5eLo28F0KgoKBAKlQVFRWhuLhYKlrVfU41Uv9OhoaGUtG9bgFe9djS0hJ2dnawsrKCtbU17Ozs7jqi88iRI5gxYwYqKyvvO7q/Li0tLXzxxRcsllKDzJ49GxcvXkRcXBy0tbXljtNmVFRUICcnB5mZmcjOzpY2VRFd1Q7X/bm8vLzeceq2yXU7/eoWnk1NTWFkZAQDAwOp/dTU1MSNGzeQmJiIefPmQUtLS/o2kLa2NkxMTKRzNGRU+p0aMho+Pz8fwP++taTqGKj77SdV+1xcXCx18N7ZsXBn5+/dmJiYSIV3VRtsaWkp/WxtbQ0rKyvY29tLj/n5b1zHjx/HihUrEB0djWeffRbvvfceunfvLncsIiIiIiJqHiy0U8tSXV2NzMxMpKen48aNG8jKykJmZiZycnKkgrrqcd1irJaWFqysrNQKDHULvvcqApuamt53ShZSV1JSgqKionodGPfr4Lh9+zby8vLUjmNgYAArKyupAJ+RkYHo6GgIIaClpQUA0NDQkDom6u6vq7q6GkOHDsWff/7Z9BdPbcKVK1fQp08f/PDDD5gxY4bccVqFiooK3LhxA+np6dKWk5MDhUIBhUIhtc13FoCNjY1ha2ur1g7fud2539TUFAYGBo+UVwhxz07NtqK4uBhFRUX37cSou+Xm5iI7OxsVFRXSMTQ0NKQOUBsbG6k9trOzQ8eOHeHo6AhHR0fY29uzIN8AQgjs27cPb7/9trRo6gcffICuXbvKHY2IiIiIiJoWC+3UvG7fvo2UlBTcuHEDN2/elIo3GRkZuHHjBhQKhTQVi7a2NmxtbdVG39na2kqP7ezspMdWVlbQ1OTavi1VVVUVcnJykJ2dDYVCIY10VSgUiI6ORmJiIsrKylBRUYGysjLpfZqamtJUCh06dICJiYk0mtXU1BTm5ubQ19fH2rVrOW0CPRBfX18kJycjOjqabcb/y83NRXJystQeX79+Xa2wnp2dLb3WwMAAnTp1go2NjfQNlbsVam1sbB65YE6Nr6ioSPoG2J0dJbdu3UJOTg4yMjKgUCikzmwtLS3Y2dnByckJDg4O0ta5c2c4OjqiW7du0pRk9D+1tbX45ZdfsHr1aly/fh3z5s3DO++8A3t7e7mjERERERFR02ChnRpffn4+UlNTpe3KlSuIi4tDSkoKCgoKpNdZWFigS5cusLOzg729fb3Hjo6OHEXXDlVVVeH27dvIyspCamoqMjMz6z1OS0uTOmRUn6O7bc7Ozm1+ZCs9uMjISHh4eODAgQPw9vaWO06zurNdrts+Z2VlSa+7W7tc9+fOnTuzg6KdUH1m6rbBD9oOu7q6ws3NDS4uLu1+seGqqirs3LkT69evR15eHhYtW
```

Comment:
Can you add extend to the TaskGraph API? Otherwise you're programming to implementation instead of interfaces. I.e. you're directly loading the yaml and working with the task list. You shouldn't be doing that IMO.

Reply:
good point. it is done and the notebook is edited accordingly.